### PR TITLE
NPM Publish Workflow

### PR DESCRIPTION
**Pull request by rryter**
- **What is the current behavior?**
Nothing happens 🤦 

- **What is the new behavior (if this is a feature change)?**
 Github Release will trigger an `npm publish --access public` to the `npm` registry. 🚀

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -26,10 +26,8 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
+          cache: 'npm'
       - run: npm ci
-      - run: git submodule update --init --recursive
-      - run: cd ./submodules/universalprofile-smart-contracts && npm ci && cd ./submodules/ERC725/implementations && npm ci
-      - run: cd ./submodules/universalprofile-smart-contracts && npx truffle build
       - run: npm run build --if-present
```
Comment (Hugoo) [inline .github/workflows/node.js.yml:31]: No lint check? I also see no 'lint' script in the package.

This action is cool, the output is better than just running 'npm run lint'

```yaml
- name: Lint
        uses: wearerequired/lint-action@v1
        with:
          eslint: true
          eslint_extensions: ts
```

**Review by Hugoo — APPROVED** [review]
cool

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        include:
+          - registry-url: 'https://registry.npmjs.org'
+            command: npm publish --access public
+            token: NPM_TOKEN
```
Comment (CallumGrindle) [inline .github/workflows/publish.yml:16]: Is this a secret? Should it be `${{ secrets.NPM_TOKEN }}`?

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (CJ42) [reply .github/workflows/node.js.yml:31]: There is also this super_linter github action. That combine multiple linters in one. If useful.
https://github.com/github/super-linter

File: .github/workflows/publish.yml (hunk 2 above)
Comment (rryter) [reply .github/workflows/publish.yml:16]: It's actually not possible to pass in secretes into the matrix. It's a workaround, to access the secret later down the road.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          node-version: '12.x'
+          registry-url: ${{ matrix.registry-url }}
+          scope: '@lukso'
+          cache: 'npm'
+      - run: npm ci
+      - run: npm run build
+      - run: ${{ matrix.command }}
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets[matrix.token] }}
```
Comment (rryter) [inline .github/workflows/publish.yml:33]: @CallumGrindle check this out :)

**Comment by rryter** [comment]
I'll look into the gh-action for the linter when I get a chance.

File: .github/workflows/publish.yml (hunk 3 above)
Comment (CallumGrindle) [reply .github/workflows/publish.yml:33]: Ah cool! Nice work around

**Pull request merged**